dev/service-registry/service-locator/src/test/scala/com/lightbend/lagom/gateway/AkkaHttpServiceGatewaySpec.scala

```diff
@@ -83,6 +87,22 @@ class AkkaHttpServiceGatewaySpec extends WordSpec with Matchers with BeforeAndAf
       response.status.intValue() should ===(404)
     }
 
+    "Rewrite 'Host: ' and stack into 'X-Forwarded-Host: ' " in {
+      val answer: String = Await.result(for {
+        response <- http.singleRequest(HttpRequest(uri = s"$gatewayUri/echo-headers"))
+        data <- response.entity.dataBytes.runFold(ByteString.empty)(_ ++ _)
+      } yield data.utf8String, 10.seconds)
+
+      val port = gateway.address.getPort
+
+      val headers = answer.split("\n")
+      headers should contain(s"Host: localhost:$servicePort")
+
+      // The following two assertions should be switched when https://github.com/akka/akka-http/issues/2191 is fixed
+      headers should contain(s"X-Forwarded-Host: localhost")
+      //      answer should contain(s"X-Forwarded-Host: localhost:$port")
+    }
+
```

Comment:
I didn't add a test for Netty because of the cost of creating the test from scratch and the plans to discontinue the Netty service gateway. Can do on a separate PR.